tests/test_core.py

```diff
@@ -413,3 +414,56 @@ def test_partial_aggregation(input_data, exp_data, warning, caplog):
         assert "WARNING" not in caplog.text
     else:
         assert all(c in caplog.text for c in warning)
+
+
+@pytest.mark.parametrize(
+    "input_data, difference",
+    [
+        (  # Variable is available in provided and aggregated data but different
+            [
+                ["m_a", "s_a", "region_A", "Primary Energy", "EJ/yr", 1, 2],
+                ["m_a", "s_a", "region_B", "Primary Energy", "EJ/yr", 3, 4],
+                ["m_a", "s_a", "World", "Primary Energy", "EJ/yr", 5, 6],
+            ],
+            [
+                ["m_a", "s_a", "World", "Primary Energy", "EJ/yr", 2005, 5, 4, 20],
+            ],
+        ),
+        (  # Conflict between overlapping renamed variable and provided data
+            [
+                ["m_a", "s_a", "region_A", "Variable B", "EJ/yr", 1, 2],
+                ["m_a", "s_a", "region_B", "Variable B", "EJ/yr", 3, 4],
+                ["m_a", "s_a", "World", "Variable B", "EJ/yr", 4, 6],
+            ],
+            [
+                ["m_a", "s_a", "World", "Variable B", "EJ/yr", 2010, 6, 4, 100.0 / 3.0],
+                ["m_a", "s_a", "World", "Variable B", "EJ/yr", 2005, 4, 3, 25],
+            ],
+        ),
+    ],
+)
+def test_aggregation_differences_export_to_file(
+    input_data, difference, cleanup_diff_file
+):
+
+    test_df = IamDataFrame(pd.DataFrame(input_data, columns=IAMC_IDX + [2005, 2010]))
+    dsd = DataStructureDefinition(TEST_DATA_DIR / "region_processing/dsd")
+    processor = RegionProcessor.from_directory(
+        TEST_DATA_DIR / "region_processing/partial_aggregation", dsd
+    )
+    processor.apply(test_df, export_difference=True)
+    columns = [
+        "model",
+        "scenario",
+        "region",
+        "variable",
+        "unit",
+        "year",
+        "original",
+        "aggregated",
+        "relative difference (%)",
+    ]
+    exp = pd.DataFrame(difference, columns=columns)
```

Comment:
Why is that necessary?

Reply:
It's to check that the differences are exported correctly.

Reply:
Sorry, I meant - is it necessary to specify the columns explicitly?

Reply:
Ah, got it, thanks!